services/src/main/java/org/keycloak/organization/validator/OrganizationMemberValidator.java

```diff
@@ -73,54 +75,81 @@ public boolean isSupported(Scope config) {
     }
 
     private void validateEmailDomain(String email, String inputHint, ValidationContext context, OrganizationModel organization) {
-        if (UserModel.EMAIL.equals(inputHint)) {
-            if (StringUtil.isBlank(email)) {
-                context.addError(new ValidationError(ID, inputHint, "Email not set"));
-                return;
-            }
-
-            if (!emailValidator().validate(email, inputHint, context).isValid()) {
-                return;
-            }
-
-            UserProfileAttributeValidationContext upContext = (UserProfileAttributeValidationContext) context;
-            AttributeContext attributeContext = upContext.getAttributeContext();
-            UserModel user = attributeContext.getUser();
-            String emailDomain = email.substring(email.indexOf('@') + 1);
-            List<String> expectedDomains = organization.getDomains().map(OrganizationDomainModel::getName).toList();
-
-            if (UserProfileContext.IDP_REVIEW.equals(attributeContext.getContext())) {
-                KeycloakSession session = attributeContext.getSession();
-                BrokeredIdentityContext brokerContext = (BrokeredIdentityContext) session.getAttribute(BrokeredIdentityContext.class.getName());
-
-                if (brokerContext != null) {
-                    String alias = brokerContext.getIdpConfig().getAlias();
-                    IdentityProviderModel broker = organization.getIdentityProviders().filter((p) -> p.getAlias().equals(alias)).findAny().orElse(null);
-
-                    if (broker == null) {
-                        return;
-                    }
-
-                    String brokerDomain = broker.getConfig().get(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
-
-                    if (brokerDomain == null) {
-                        return;
-                    }
-
-                    expectedDomains = List.of(brokerDomain);
-                }
-            } else if (!organization.isManaged(user)) {
-                return;
-            }
-
-            if (expectedDomains.isEmpty()) {
-                return;
-            }
-
-            if (!expectedDomains.contains(emailDomain)) {
-                context.addError(new ValidationError(ID, inputHint, "Email domain does not match any domain from the organization"));
-            }
+        if (!UserModel.EMAIL.equals(inputHint)) {
+            return;
+        }
+
+        if (StringUtil.isBlank(email)) {
+            context.addError(new ValidationError(ID, inputHint, "Email not set"));
+            return;
+        }
+
+        if (!emailValidator().validate(email, inputHint, context).isValid()) {
```

Comment:
I think I missed this when we introduced this validator, but shouldn't this situation add an error to the context?

Reply:
It will add one. See `EmailValidator`.